### PR TITLE
feat: make saved filter views UI mobile friendly

### PR DESCRIPTION
**Pull request by xMort**
JIRA: LX-421
risk: low

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/filterViews/FilterViews.tsx [hunk 1]
```diff
@@ -1,64 +1,141 @@
...
+
+    return {
+        toggleDialog,
+        openListDialog,
+        openAddDialog,
+        closeDialog,
+    };
+};
 
+export const FilterViews: React.FC = () => {
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/filterViews/FilterViews.tsx:85]: Component open / mode state is in redux now so component can be managed from mobile menu dropdown.

File: libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/useDefaultMenuItems.tsx [hunk 2]
```diff
@@ -114,137 +159,143 @@ export function useDefaultMenuItems(): IMenuButtonItem[] {
 
     const menuButtonItemsVisibility = useDashboardSelector(selectMenuButtonItemsVisibility);
 
+    const isFilterViewsFeatureFlagEnabled = useDashboardSelector(selectEnableFilterViews);
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/useDefaultMenuItems.tsx:162]: Refactored to fix complexity issue reported by Sonar.

File: libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/useDefaultMenuItems.tsx [hunk 3]
```diff
@@ -114,137 +159,143 @@ export function useDefaultMenuItems(): IMenuButtonItem[] {
...
+                {
+                    type: "button",
+                    itemId: "schedule-email-edit-item", // careful, this is also used as a selector in tests, do not change
+                    itemName: intl.formatMessage({ id: "options.menu.schedule.email.edit" }),
+                    onClick: defaultOnScheduleEmailingManagement,
+                    visible: isScheduledManagementEmailingVisible,
+                    icon: "gd-icon-list",
+                },
+            ],
+            // filter views section
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/useDefaultMenuItems.tsx:238]: These two items are new.

File: libs/sdk-ui-filters/src/DateFilter/Select/Select.tsx [hunk 4]
```diff
@@ -66,11 +69,3 @@ export const Select = <V extends {}>({
         </TypedDownshift>
     );
 };
-
-Select.defaultProps = {
```
Comment (xMort) [inline libs/sdk-ui-filters/src/DateFilter/Select/Select.tsx:70]: These are reported as red warning in browser console as soon to be deprecated by future version of react. It made my console unreadable so I got rid of them.

**Review by NestorEncinas — APPROVED** [review]

**Pull request merged**